### PR TITLE
roachtest: patch and update pg_regress roachtest

### PR DESCRIPTION
**Pull request by rharding6373**
This PR adds patches to apply to the pg_regress test suite to fix some non-deterministic ordering in CRDB and avoid running queries that result in internal errors, since the latter can be noisy.

Epic: none

Release note: none

### CODE REVIEW

**Review by yuzefovich — APPROVED** [review]
:lgtm: (although it's the first time I looked into this roachtest, so perhaps someone who reviewed it originally should take a look too).

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/118717#-:-NpukpzVCUgScgGq45AI:b1y608c)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @herkolategan, @renatolabs, and @rharding6373)

___
*[`pkg/cmd/roachtest/tests/pg_regress.go` line 480 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/118717#-NpuiEZr2Ys01rsYeMej:-NpuiEZr2Ys01rsYeMek:b-5oovfh) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9422cda8472e534fe5a7ce456bcb6efaa856bda/pkg/cmd/roachtest/tests/pg_regress.go#L480)):*
> ```Go
> // Diffs on files in the original pg_regress suite that help with setup
> // and reduce non-determinism due to CRDB implementation.
> // Note: Due to issues applying patches including single quotes, replace
> ```

nit: would it be possible to do this replacement automatically, as a preprocessing step of applying a patch, rather than manually including `'"'"'` strings?

**Review by rharding6373 — COMMENTED** [review]
TFTR! I made another run today with master and captured the changes in regression.diffs. I'll make sure everything's green before merging.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/118717#-:-NpupOzq1ndLvbU36sd1:b-1qdpdz)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @herkolategan, @renatolabs, and @yuzefovich)

___
*[`pkg/cmd/roachtest/tests/pg_regress.go` line 480 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/118717#-NpuiEZr2Ys01rsYeMej:-Npup4osCsI7p0XW0XVQ:bxiyar4) ([raw file](https://github.com/cockroachdb/cockroach/blob/a9422cda8472e534fe5a7ce456bcb6efaa856bda/pkg/cmd/roachtest/tests/pg_regress.go#L480)):*
<details><summary><i>Previously, yuzefovich (Yahor Yuzefovich) wrote…</i></summary><blockquote>

nit: would it be possible to do this replacement automatically, as a preprocessing step of applying a patch, rather than manually including `'"'"'` strings?
</blockquote></details>

Let me look into this async. This is definitely annoying and it's not immediately obvious why the patch doesn't apply if this is missing, so it would be nice to automate in some way.

**Pull request merged**
